### DISCUSSION TITLE
Tracking down "No job found named Shiny.Notifications.NotificationJob"

### DISCUSSION THREAD

**Discussion by munkii** [shinyorg/shiny#1528] [Q&A] [answered]
I am seeing a lot of the following handled Exception being logged in AppCenter

```
AbstractJobManager.Run (System.String jobName, System.Threading.CancellationToken cancelToken)
System.ArgumentException: No job found named Shiny.Notifications.NotificationJob, Shiny.Notifications, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
```

We started using Shiny.Jobs (3.2.4) and since that build has been in the wild I have been seeing these Exceptions in AppCenter. I cannot see were that Run method is being called to try and run a NotificationJob class that does not even exist.

Could it be an old scheduled Notification from a previously build and the Device is still trying to run it or am I just talking crazy?

This does not happen in our DEV and TEST builds but those devices\builds are not normal in that we are always uninstalling, upgrading or fresh installing on them

**Comment by aritchie** [comment, marked as answer]
It's an old left over job from notifications that has been removed.  The job engineer does try to clean these missing types out

**Reply by munkii** [reply]
One thing that these Exceptions did highlight is that whilst they are being logged in AppCenter as "Errors" rather than "Crashes" so they are being handled I never see them in Sentry.

I realise that this is not a Shiny issue but I know you've used Sentry and AppCenter yourself. Have you ever seen this discrepancy? We see lots of other known\expected Exceptions in Sentry.

**Reply by aritchie** [reply]
No.  Shiny uses the general MSFT logging which sentry plugs into so it should be the same in both products.
